### PR TITLE
Test to catch a bug in builtins.match

### PR DESCRIPTION
**Pull request by mightybyte**
Also added infrastructure for comparing against current nix behavior,
not just current nix tests.

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -1,32 +1,7 @@
...
-  haskellPackages = let hpkgs = pkgs.haskell.packages.${compiler}; in
-    hpkgs // {
-      mkDerivation = args: hpkgs.mkDerivation (args // {
-        enableLibraryProfiling = doProfiling;
-        enableExecutableProfiling = doProfiling;
-      });
-    };
-
-  pkg = haskellPackages.developPackage {
+{ pkgs ? import <nixpkgs> {} }: pkgs.haskellPackages.developPackage {
```
Comment (jwiegley) [inline default.nix]: This change loses the pinning, and the ability to enable profiling. DId you overwrite default.nix with something you intended only to use locally?

File: tests/EvalTests.hs [hunk 2]
```diff
@@ -71,6 +72,8 @@ case_function_recursive_sets =
 case_nested_with =
     constantEqualStr "2" "with { x = 1; }; with { x = 2; }; x"
 
+case_match_failure_null = assertEvalMatchesNix "builtins.match \"ab\" \"abc\""
```
Comment (jwiegley) [inline tests/EvalTests.hs:75]: This is a great idea!

File: default.nix (hunk 1 above)
Comment (mightybyte) [reply default.nix]: Ack!  Yes.  Your `default.nix` file does not work for me.  Let me change it back.

**Comment by jwiegley** [comment]
Before we install this, we have to decide what to do about Travis. Can we install Nix there?  Do you know @ryantrinkle?

**Comment by mightybyte** [comment]
Yeah, we're getting this CI failure:

```
nix-instantiate: readCreateProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
```

I knew this was a possibility.  I was thinking we should probably add `nix` to `testHaskellDepends`.

**Comment by jwiegley** [comment]
You can `nix` to `testHaskellDepends` in your PR, and Travis will give it a try.

**Comment by jwiegley** [comment]
Merged manually.

**Pull request closed without merging**